src/react-components/room/TipContainer.js

```diff
@@ -16,8 +16,8 @@ let turnRightKey = "E";
 if (window.navigator.keyboard !== undefined && window.navigator.keyboard.getLayoutMap) {
   window.navigator.keyboard.getLayoutMap().then(function(map) {
     moveKeys = `${map.get("KeyW")} ${map.get("KeyA")} ${map.get("KeyS")} ${map.get("KeyD")}`.toUpperCase();
```

Comment:
If these return `null` we'll get `moveKeys = "NULL NULL NULL NULL"`, but I guess if we're on mobile, `moveKeys` won't be used in a tip... Still, perhaps this line should read
```js
    moveKeys = `${map.get("KeyW") || "W"} ${map.get("KeyA") || "A"} ${map.get("KeyS") || "S"} ${map.get("KeyD") ||
      "D"}`.toUpperCase();
```

Reply:
I've added that in as suggested. I ignore it originally because, as you say, it's not used, but it could be exposed and trip somebody up in the future.